src/metrics.py

```diff
@@ -0,0 +1,148 @@
+#NLTK is standard and comes with python
+from nltk.translate.bleu_score import corpus_bleu,sentence_bleu
+from nltk.translate.meteor_score import meteor_score
+from nltk import word_tokenize
+import nltk
+nltk.download('wordnet')
+#evaluate also has all BLEU and METEOR scores if standard nltk doesn't work
+import evaluate
+from aac_metrics.functional import cider_d
+from aac_metrics.utils.tokenization import preprocess_mono_sents, preprocess_mult_sents
+import subprocess
+
+'''
+This function calculates the BLEU-4 score for an entire corpus. That means we would have a list of all the captions in references and a list of all the predicted captions in candidates.
+'''
+def calculate_bleu_score_corpus(references, candidates):    
+    #Example: [[ref1a sentence, ref1bsentence], [ref2a sentence, ref2b sentence]]
+    #[['The cat is on the mat', 'there is a cat on the wall'],['The cat is not on the mat', 'there is a dog on the wall']]
+
+    #Example: [cand1 sentence, cand2 sentence]
+    #['The cat is on the mat','The cat is not on the mat']
```

Comment:
Try to follow standard documentation format for function docs, like this for ex: https://github.com/farazali7/multimodal-xray/blob/99a267faf6980217b7308021e09c6aa9441d06e3/src/models/attention.py#L11